Script to ease config of GPG/SSH on a new computer. (#84)
DataDog/yubikey: docs/optional.md

# Optional

- [Configure another computer to use a configured YubiKey](#configure-another-computer-to-use-a-configured-yubikey)
- [Signing for different git repositories with different keys](#signing-for-different-git-repositories-with-different-keys)
- [Keybase](#keybase)
- [VMware Fusion](#vmware-fusion)
- [Docker Content Trust](#docker-content-trust)

## Configure another computer to use a configured YubiKey

You don't need to do anything extra if you have not set up GPG and SSH to your use YubiKey.

Otherwise, you need to:

> 1. Get a copy of your Yubikey GPG public key, this should be backed up in your 1password personal vault

Author: ```suggestion
1. Get a copy of your Yubikey GPG public key, this should have been backed up in your password manager
```

2. Get the Yubikey GPG key ID by running `gpg --list-keys`, in the following example the key ID is `4E09860E71D948019BD426D5D099A306DBECDF1B`  
![image](https://user-images.githubusercontent.com/4062883/148108677-ab3a04b4-8ef6-4ba0-b78e-ec9c127857e3.png)

4. Write your copy of your GPG public key stored in your password manager to disk.
5. Run [`./import.sh -p /path/to/pubkey.asc -i key_id`](../import.sh).
6. You will be prompted several times:

>     1. for installing dependencies, type yes, and press enter

Author: ```suggestion
    1. To install dependencies (required), type yes, and press enter
```

>     2. to know if you want to configure the Yubikey GPG key for commit signing, type yes or no, and press enter

Author: ```suggestion
    2. To configure the Yubikey GPG key for commit signing (or not), type yes or no, and press enter
```

>     3. to know if you want to configure the Yubikey SSH key for SSH connections, type yes or no, and press enter

Author: ```suggestion
    3. To use the Yubikey GPG key for SSH connections (or not), type yes or no, and press enter
```
